### PR TITLE
Add support for paginated workflow history

### PR DESCRIPTION
**Pull request by antstorm**
Cadence paginates history when it becomes too big, this PR enables the gem to prefetch all of it before processing a decision task.

Also added a full spec for DecisionTaskProcessor along with this a spec for this change

### CODE REVIEW

**Comment by jordanwnolan** [comment]
Overall looks good to me. Is there any concern about the history getting too large causing memory problems?

**Comment by antstorm** [comment]
@jordanwnolan memory as in RAM on worker or DB size on a cluster?

File: lib/cadence/client/thrift_client.rb [hunk 1]
```diff
@@ -97,13 +97,14 @@ def start_workflow_execution(
...
-      def get_workflow_execution_history(domain:, workflow_id:, run_id:)
+      def get_workflow_execution_history(domain:, workflow_id:, run_id:, next_page_token: nil)
         request = CadenceThrift::GetWorkflowExecutionHistoryRequest.new(
           domain: domain,
           execution: CadenceThrift::WorkflowExecution.new(
             workflowId: workflow_id,
             runId: run_id
-          )
+          ),
+          nextPageToken: next_page_token
```
Comment (cedric-cordenier) [inline lib/cadence/client/thrift_client.rb:107]: is mixing the case intentional here?

Comment (anthonydmitriyev) [reply lib/cadence/client/thrift_client.rb:107]: @cedric-cordenier yes, Cadence Thrift definitions all use camelCase :/

**Review by cedric-cordenier — APPROVED** [review]

**Pull request merged**